scripts/frame/main.js

```diff
@@ -125,6 +141,21 @@ frame.set = function (data) {
 	frame.refreshPicture();
 };
 
+frame.resize = function() {
```

Comment:
why do you need this ? This should be handled by the CSS. Is it not ?

Reply:
CSS handles the layout but is _not_ used for determining which image variant from `srcset` is to be used. For that we need to provide the browser with a (relatively) accurate rendered image width in the `sizes` attribute; and the image width can obviously change when the viewport is resized, so we recalculate. We have the same sort of code for the regular photo view, BTW (in `view.photo.onresize()`).

At the risk of talking about things you may be perfectly familiar with, allow me to expand as it was something that surprised _me_ about `srcset` when I was reading the specs and experimenting with it in the browsers. Basically, we provide the URLs and widths of the image variants in the `srcset` attribute, but that's not enough for the browser's rendering engine. The browser also needs the expected rendered width to be provided, via `sizes`. Apparently that's because images are loaded before the browser determines the page layout, so at the time they are loaded the browser doesn't know what the width of the rendered image is going to be. So by default it assumes the worst-case scenario of `100vw`, potentially resulting in the loading of a larger image variant than is actually needed. E.g., in case of the frame mode displayed full-screen on a 16:9 display, images in portrait will use less than half of the viewport width. The browser doesn't know it without loading the image first, but we do because we have the width/height information in the database so we have a good idea of how wide it will be rendered. That's the purpose of the math that the code does here.